tck/src/test/java/org/reactivestreams/tck/RangePublisherTest.java

```diff
@@ -0,0 +1,176 @@
+/************************************************************************
+ * Licensed under Public Domain (CC0)                                    *
+ *                                                                       *
+ * To the extent possible under law, the person who associated CC0 with  *
+ * this code has waived all copyright and related or neighboring         *
+ * rights to this code.                                                  *
+ *                                                                       *
+ * You should have received a copy of the CC0 legalcode along with this  *
+ * work. If not, see <http://creativecommons.org/publicdomain/zero/1.0/>.*
+ ************************************************************************/
+
+package org.reactivestreams.tck;
+
+import java.util.Map;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.concurrent.atomic.*;
+
+import org.reactivestreams.*;
+import org.testng.annotations.*;
+
+@Test
+public class RangePublisherTest extends PublisherVerification<Integer> {
+
+    static final Map<Integer, StackTraceElement[]> stacks = new ConcurrentHashMap<Integer, StackTraceElement[]>();
+    
+    static final Map<Integer, Boolean> states = new ConcurrentHashMap<Integer, Boolean>();
+    
+    static final AtomicInteger id = new AtomicInteger();
+    
+    @AfterClass
+    public static void afterClass() {
+        boolean fail = false;
+        StringBuilder b = new StringBuilder();
+        for (Map.Entry<Integer, Boolean> t : states.entrySet()) {
+            if (!t.getValue()) {
+                b.append("\r\n-------------------------------");
+                for (Object o : stacks.get(t.getKey())) {
+                    b.append("\r\nat ").append(o);
+                }
+                fail = true;
+            }
+        }
+        if (fail) {
+            throw new AssertionError("Cancellations were missing:" + b);
+        }
+    }
+    
+    public RangePublisherTest() {
+        super(new TestEnvironment(25));
```

Comment:
We use 25 ms for most of the synchronous tests in RxJava because otherwise it takes minutes to work through the 100+ TCK tests. Since this Publisher is synchronous, there is little to no chance for a timeout in my opinion.